### ISSUE TITLE
Support for Open and RADIUS Servers

### ISSUE THREAD

**Issue by Beardwin1** [CiscoDevNet/ansible-meraki#296]
Hi,

At the moment you can't have auth_mode set to open and use the radius_servers keyword, as radius_servers requires open_with_radius or 802.1x radius, would it be possible to address this so that we can use Ansible for automating guest networks that require this specific setting, would be much appreciated! Love your hard work btw :).

This is the error I am faced with when setting auth_mode to open and using the radius_servers keyword in the playbook:
**fatal: [localhost]: FAILED! => {"changed": false, "msg": "radius_servers requires auth_mode to be open-with-radius or 8021x-radius", "response": null, "status": null}**

This is the play in my playbook if you wanted to take a look:

**- name: Company Name

meraki_ssid:

auth_key: #####################

state: present

org_name: #########

net_name: #########

name: Company Name

enabled: true

auth_mode: open

splash_page: Password-protected with custom RADIUS

radius_proxy_enabled: yes

radius_servers:

- host: ###############

port: 1812

secret: ############

- host: ###############

port: 1812

secret: ############

radius_accounting_enabled: yes

radius_accounting_servers:

- host: ###############

port: 1813

secret: ############

- host: ###############

port: 1813

secret: ############

ip_assignment_mode: NAT mode

delegate_to: localhost**

Meraki does support this configuration as per the below:
![image](https://user-images.githubusercontent.com/86856010/161890338-df9dc02b-3eb4-47ed-bf7d-527f6375e915.png)

This is the documentation I was following:
https://docs.ansible.com/ansible/latest/collections/cisco/meraki/meraki_mr_ssid_module.html#ansible-collections-cisco-meraki-meraki-mr-ssid-module

**Comment by kbreit** [comment]
I'm looking into this one more and the notes from Meraki do indeed say

> The RADIUS 802.1X servers to be used for authentication. This param is only valid if the authMode is 'open-with-radius', '8021x-radius' or 'ipsk-with-radius'

I'm asking someone at Meraki about correct behavior and will report back. Either way, I can add `ipsk-with-radius` as an option.

**Comment by Beardwin1** [comment]
> I'm looking into this one more and the notes from Meraki do indeed say
> 
> > The RADIUS 802.1X servers to be used for authentication. This param is only valid if the authMode is 'open-with-radius', '8021x-radius' or 'ipsk-with-radius'
> 
> I'm asking someone at Meraki about correct behavior and will report back. Either way, I can add `ipsk-with-radius` as an option.

Thanks for the reply. If possible I'd want the encryption set to "open" and "radius for splash page" options. Understand if it's out of your control and thanks a lot for your effort! Will await a response. Do let me know when they get back to you.

**Comment by kbreit** [comment]
What are you trying to accomplish by having it be open and have a backend RADIUS server? If it's open there's inherently no authorization or authentication so the RADIUS server wouldn't have a place.

**Comment by TKIPisalegacycipher** [comment]
First, could you confirm this is a valid config Meraki allows via GUI?

If so, please provide screenshots of the config elements.

**Comment by Beardwin1** [comment]
> Just saw your screenshots above.
> 
> Configuring that yields a config similar to this. Are you able to update an SSID using these params?
> 
> I think only relevant params here are splashPage, authMode, radiusServers
> 
> ```
>     {
>         "number": 0,
>         "name": "Default Meraki configuration - w",
>         "enabled": true,
>         "splashPage": "Password-protected with custom RADIUS",
>         "ssidAdminAccessible": false,
>         "authMode": "open",
>         "radiusServers": [
>             {
>                 "host": "1.1.1.1",
>                 "port": 123,
>                 "id": "797137134044981750",
>                 "openRoamingCertificateId": null,
>                 "caCertificate": null
>             }
>         ],
>         "radiusTestingEnabled": false,
>         "radiusFailoverPolicy": "Deny access",
>         "radiusLoadBalancingPolicy": "Strict priority order",
>         "ipAssignmentMode": "NAT mode",
>         "adultContentFilteringEnabled": false,
>         "dnsRewrite": {
>             "enabled": false,
>             "dnsCustomNameservers": []
>         },
>         "adminSplashUrl": null,
>         "splashTimeout": "1440 minutes",
>         "walledGardenEnabled": false,
>         "minBitrate": 11,
>         "bandSelection": "Dual band operation",
>         "perClientBandwidthLimitUp": 0,
>         "perClientBandwidthLimitDown": 0,
>         "perSsidBandwidthLimitUp": 0,
>         "perSsidBandwidthLimitDown": 0,
>         "mandatoryDhcpEnabled": false,
>         "visible": true,
>         "availableOnAllAps": true,
>         "availabilityTags": [],
>         "speedBurst": {
>             "enabled": false
>         }
>     }
> ```

Sorry not entirely sure what you want me to do? Could you explain a bit more?

**Comment by TKIPisalegacycipher** [comment]
Are you able to push that config via the API (without using Ansible)?

E.g. via postman or a curl command?

**Comment by Beardwin1** [comment]
> Are you able to push that config via the API (without using Ansible)?
> 
> E.g. via postman or a curl command?

Ah thanks for that, that would probably work but I don't really have any experience with that. I just learned Ansible, so that's all I know so far trying to learn other technologies too, but still new to this field of IT.

**Comment by kbreit** [comment]
Give me a day or so and I can write a `curl` command you can run from a Linux or Mac CLI to test.

**Comment by Beardwin1** [comment]
> Give me a day or so and I can write a `curl` command you can run from a Linux or Mac CLI to test.

Thanks, really appreciate it!

**Comment by kbreit** [comment]
@Beardwin1 Obviously modify the values to match what you need. But this is a proof of concept.

```
curl -X PUT -H "X-Cisco-Meraki-API-Key: abc123" -H "Content-Type: application/json" --data '{"authMode": "open","radiusServers": [{"host": "1.1.1.1","port": 123,"secret": "abc123"}],"splashPage": "Password-protected with custom RADIUS","radiusFailoverPolicy": "Allow access","radiusLoadBalancingPolicy": "Strict priority order"}' https://api.meraki.com/api/v1/networks/N_123456/wireless/ssids/1
```

@TKIPisalegacycipher FWIW, it worked for me with `curl`.

**Comment by Beardwin1** [comment]
> @Beardwin1 Obviously modify the values to match what you need. But this is a proof of concept.
> 
> ```
> curl -X PUT -H "X-Cisco-Meraki-API-Key: abc123" -H "Content-Type: application/json" --data '{"authMode": "open","radiusServers": [{"host": "1.1.1.1","port": 123,"secret": "abc123"}],"splashPage": "Password-protected with custom RADIUS","radiusFailoverPolicy": "Allow access","radiusLoadBalancingPolicy": "Strict priority order"}' https://api.meraki.com/api/v1/networks/N_123456/wireless/ssids/1
> ```
> 
> @TKIPisalegacycipher FWIW, it worked for me with `curl`.

Thanks a lot! Just to check, I need to adjust the values to what I need and just copy and paste it and run it on Linux right?

**Comment by kbreit** [comment]
> > @Beardwin1 Obviously modify the values to match what you need. But this is a proof of concept.
> 
> > 
> 
> > ```
> 
> > curl -X PUT -H "X-Cisco-Meraki-API-Key: abc123" -H "Content-Type: application/json" --data '{"authMode": "open","radiusServers": [{"host": "1.1.1.1","port": 123,"secret": "abc123"}],"splashPage": "Password-protected with custom RADIUS","radiusFailoverPolicy": "Allow access","radiusLoadBalancingPolicy": "Strict priority order"}' https://api.meraki.com/api/v1/networks/N_123456/wireless/ssids/1
> 
> > ```
> 
> > 
> 
> > @TKIPisalegacycipher FWIW, it worked for me with `curl`.
> 
> 
> 
> Thanks a lot! Just to check, I need to adjust the values to what I need and just copy and paste it and run it on Linux right?

Correct. Network name and API key are the most obvious. Probably the SSID number too. If you need help getting those let me know.
